src/lib.rs

```diff
@@ -429,7 +429,7 @@ where
                             #[cfg(not(any(feature = "tls", feature = "rustls")))]
                             Some(_) => panic!("hyper-proxy was not built with TLS support"),
 
-                            None => Ok(ProxyStream::Regular(tunnel_stream)),
+                            None => Ok(ProxyStream::NoProxy(tunnel_stream)),
```

Comment:
I think this should still be ProxyStream::Regular, shouldn't it? You've still matched a proxy here, you just don't have any TLS support enabled.

Reply:
fixed

Reply:
Looks good.